lib/dynamics_crm/client.rb

```diff
@@ -167,6 +181,35 @@ def delete(entity_name, guid)
       return Response::DeleteResponse.new(xml_response)
     end
 
+    def bulk_delete(entity_name, emailToRecipientGuid, criteria=[], wait=false)
+
+      query = XML::Query.new(entity_name)
+      query.criteria = XML::Criteria.new(criteria)
+
+      request = bulk_delete_request(query, emailToRecipientGuid)
+      xml_response = post(organization_endpoint, request)
+
+      response = Response::ExecuteResult.new(xml_response)
+      jobId = response['JobId']
+      if wait && !jobId.nil?
+        while true
+          jobStatus = check_job_status(jobId)
+          if jobStatus != "Waiting For Resources" && jobStatus != "Waiting" && jobStatus != "In Progress"
+            break
+          end
+          sleep(2) # wait x seconds and check again
+        end
+
+        if jobStatus != 'Succeeded'
+          response['errorMessage'] = "The job has not completed with success. The response returned a status of #{jobStatus}. Please check your query/parameters and try again."
+        end
+
+        return response
+      else
```

Comment:
Done.

Travis CI build failed, though. Not sure why, this time (seems to be not related with bundler version, as in case above).